packages/mjml-core/src/helpers/postRender.js

```diff
@@ -1,4 +1,4 @@
-export const removeCDATA = str => str.replace(/<!--\[CDATA\[([^]*?)\]\]-->/gm, '$1')
+export const removeCDATA = str => str.replace(/<!--\[CDATA\[([^]*?)]]-->/gm, '$1')
```

Comment:
Since you never actually opened the character class (due to the escape), the closing square bracket does not have a special meaning and therefore does not need escaping